armi/reactor/reactors.py

```diff
@@ -293,15 +292,26 @@ def powerMultiplier(self):
 
     @property
     def lib(self):
-        """"Get the microscopic cross section library."""
-        if self._lib is None:
-            runLog.info("Loading microscopic cross section library ISOTXS")
+        """
+        Return the microscopic cross section library.
+
+        Notes
+        -----
+        This returns the current library object (default as ``ISOTXS``) stored on
+        the core state. If no library has been set on the core yet, this will
+        load in an existing ``ISOTXS`` file from the working directory if one
+        exists. If an ``ISOTXS`` file does not exist in the working directory
+        no library will be loaded.
+        """
+        isotxsFileName = nuclearDataIO.getExpectedISOTXSFileName()
+        if self._lib is None and os.path.exists(isotxsFileName):
+            runLog.info(f"Loading microscopic cross section library `{isotxsFileName}`")
             self._lib = nuclearDataIO.ISOTXS()
```

Comment:
this needs to use the expected name

Reply:
I also feel like this should crash if the file isn't there. Can you explain what scenario needs this?

Reply:
There simply is no guarantee that a Core has a lib, or can get one. If this just blows up when anyone asks, then the only way to check _whether_ it has one is in a try/except, which is not ideal. Code that needs a lib is welcome to pitch a fit, and code that wants one can check `is not None:` and roll with it if it doesn't exist.